lib/withHotKeys.js

```diff
@@ -0,0 +1,76 @@
+import React, {PureComponent} from 'react';
+import HotKeys from './HotKeys';
+
+/**
+ * withHotKeys is an HOC that provides the wrappedComponent with the ability to implement keyboard actions
+ * without the user wrapping every component with a <HotKeys> component individually
+ *
+ * See examples/master/HOCWrappedNode.js for an example implementation
+ * Follow the steps below to use the HOC:
+ *
+ * @example <caption>Example usage of withHotKeys.</caption>
+ * // Returns the HOC-wrapped component.
+ * // 1. Declared a key map that with the actionName as key and keyboardKeys as values
+ * const ACTION_KEY_MAP = {
+ *     'logConsole' : 'down',
+ * };
+ *
+ * class BasicBox extends React.Component {
+ *
+ * // 2. declare 'hotKeyHandlers' within the Component's class definition
+ *   hotKeyHandlers: {
+ *     'logConsole': this.logConsole.bind(this),
+ *   }
+ *
+ *   logConsole() {
+ *     console.log('a hotkey is pressed');
+ *   }
+ *
+ *   render() {
+ *     return (
+ *         <div tabIndex="0">
+ *             Press the down arrow
+ *         </div>
+ *     );
+ *   }
+ * }
+ *
+ * // 3. Wrap the Component with withHotKeys
+ * export default withHotKeys(ACTION_KEY_MAP)(BasicBox);
+ * @returns {function} Returns the HOC-wrapped component.
+ *
+ * @param {Object} keyMap an action-to-keyboard-key mapping
+ * @summary An HOC that provides the wrappedComponent with the ability to implement keyboard actions
+ */
+const withHotKeys = (keyMap, handlers) => ((Component) =>
+    class withHotKeysWrapper extends PureComponent {
+      static propTypes = Component.propTypes;
+      state = {
+        handlers: {}
+      };
+
+      componentDidMount() {
+        this.setState({handlers: this._ref.hotKeyHandlers});
```

Comment:
Have you tested this in the browser? I have previously had issues when I've experimented with trying to change the parameters passed to `HotKeys` using state. I wish I could remember what happened and whether it was `handlers` or `keyMap` I was trying to change. 

In any case, I came away the (potentially erroneous) understanding that `HotKeys` didn't support changing its arguments after it had mounted.

Reply:
Yep! This was tested using the `HOCWrappedNode` in the example html

Reply:
OK, that's good enough for me. ;)